### DISCUSSION TITLE
Listen on only 1 interface as docker has loads.

### DISCUSSION THREAD

**Discussion by simonmcnair** [christgau/wsdd#205] [Q&A] [answered]
I set up my conf file as follows:
WSDD_OPTIONS="--shortlog --workgroup='MCNAIR' -i vmbr0 -i 10.150.30.240"

but it still seems to listen on all interfaces according to netstat -tulpn

Have I done it wrong pls ?

**Comment by christgau** [comment]
Can you please paste the wsdd-related output of the netstat command here? In addition, you may run wsdd manually on the command line (stop the daemon first) and add `-vv` to the options. That will show which address are going to be used.

**Comment by simonmcnair** [comment]
okay, so It isn't doing what I expected but it isn't the fault of wsdd.  In debian the following is the systemd file:
`
[Unit]
Description=Web Services Dynamic Discovery host daemon
Documentation=man:wsdd(8)
; Start after the network has been configured
After=network-online.target
Wants=network-online.target

[Service]
Type=simple
; The service is put into an empty runtime directory chroot,
; i.e. the runtime directory which usually resides under /run
EnvironmentFile=/etc/default/wsdd
ExecStart=/usr/sbin/wsdd --shortlog --chroot=/run/wsdd $WSDD_PARAMS
DynamicUser=yes
User=wsdd
Group=wsdd
RuntimeDirectory=wsdd
AmbientCapabilities=CAP_SYS_CHROOT

[Install]
WantedBy=multi-user.target
`

the file /etc/default/wsdd contains
`WSDD_OPTIONS="--shortlog --workgroup='MCNAIR' -i vmbr0 -i 10.150.30.240"`

but it didn't work.  However this does:
`
[Unit]
Description=Web Services Dynamic Discovery host daemon
Documentation=man:wsdd(8)
; Start after the network has been configured
After=network-online.target
Wants=network-online.target

[Service]
Type=simple
; The service is put into an empty runtime directory chroot,
; i.e. the runtime directory which usually resides under /run
EnvironmentFile=/etc/default/wsdd
ExecStart=/usr/sbin/wsdd --shortlog --chroot=/run/wsdd --workgroup='MCNAIR' -i vmbr0 -i 10.150.30.240
DynamicUser=yes
User=wsdd
Group=wsdd
RuntimeDirectory=wsdd
AmbientCapabilities=CAP_SYS_CHROOT

[Install]
WantedBy=multi-user.target
`

**Comment by simonmcnair** [comment]
BEFORE:
udp        0      0 10.150.30.240:3702      0.0.0.0:*                           313307/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           313307/python3
udp        0      0 172.20.0.1:3702         0.0.0.0:*                           313307/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           313307/python3
udp        0      0 172.18.0.1:3702         0.0.0.0:*                           313307/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           313307/python3
udp        0      0 172.17.0.1:3702         0.0.0.0:*                           313307/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           313307/python3
udp        0      0 172.22.0.1:3702         0.0.0.0:*                           313307/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           313307/python3
udp        0      0 172.21.0.1:3702         0.0.0.0:*                           313307/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           313307/python3
udp        0      0 172.19.0.1:3702         0.0.0.0:*                           313307/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           313307/python3
udp        0      0 100.64.162.117:3702     0.0.0.0:*                           313307/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           313307/python3
udp6       0      0 fe80::e852:e4ff:fe:3702 :::*                                313307/python3
udp6       0      0 ff02::c:3702            :::*                                313307/python3
udp6       0      0 fe80::dabb:c1ff:fe:3702 :::*                                313307/python3
udp6       0      0 ff02::c:3702            :::*                                313307/python3
udp6       0      0 fe80::8011:89ff:fe:3702 :::*                                313307/python3
udp6       0      0 ff02::c:3702            :::*                                313307/python3
udp6       0      0 fe80::ec6d:ffff:fe:3702 :::*                                313307/python3
udp6       0      0 ff02::c:3702            :::*                                313307/python3
udp6       0      0 fe80::42:89ff:fe5e:3702 :::*                                313307/python3
udp6       0      0 ff02::c:3702            :::*                                313307/python3
udp6       0      0 fe80::42:7fff:fefc:3702 :::*                                313307/python3
udp6       0      0 ff02::c:3702            :::*                                313307/python3
udp6       0      0 fe80::42:cdff:fe2c:3702 :::*                                313307/python3
udp6       0      0 ff02::c:3702            :::*                                313307/python3
udp6       0      0 fe80::e402:c67b:10:3702 :::*                                313307/python3
udp6       0      0 ff02::c:3702            :::*                                313307/python3

/etc/default# ps -aux | grep 313307
wsdd      313307  0.0  0.0  37452 12112 ?        Ss   May01   0:00 python3 /usr/sbin/wsdd --shortlog --chroot=/run/wsdd

AFTER:
root@host:/etc/default# netstat -tulpn| grep 3702
udp        0      0 10.150.30.240:3702      0.0.0.0:*                           559990/python3
udp        0      0 239.255.255.250:3702    0.0.0.0:*                           559990/python3
udp6       0      0 fe80::dabb:c1ff:fe:3702 :::*                                559990/python3
udp6       0      0 ff02::c:3702            :::*                                559990/python3

/etc/default# ps -aux | grep 313307
wsdd      559990  2.9  0.0  37452 29012 ?        Ss   18:50   0:00 python3 /usr/sbin/wsdd --shortlog --chroot=/run/wsdd --workgroup=MCNAIR -i vmbr0 -i 10.150.30.240
root      559993  0.0  0.0   6856  2048 pts/0    S+   18:50   0:00 grep wsdd

**Comment by simonmcnair** [comment, marked as answer]
aaah.  It's because the variable is called WSDD_OPTIONS instead of $WSDD_PARAMS

**Comment by simonmcnair** [comment]
I think this is the mistake of openmediavault.  I will check with them.

**Comment by simonmcnair** [comment]
It was fixed here.... https://github.com/openmediavault/openmediavault/commit/9b8d2e13552fc27fa7c59176251556921fbcc5b8
on Dec 3, 2023
